### PR TITLE
#7266 Remove usage from ConfigurationHolder

### LINKED ISSUE #7266: Remove `ConfigurationHolder` from `swirlds-common`
Remove `ConfigurationHolder` from the `swirlds-common` module.

### PR DESCRIPTION
**Pull request by timo0**
Fix #7266

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/editor/StateEditorSave.java
Comment (hendrikebbers) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/editor/StateEditorSave.java:77]: we should extract the "default config setup" in a util method

Comment (timo0) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/editor/StateEditorSave.java]: Introduced `DefaultConfiguration`

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/config/DefaultConfiguration.java
Comment (cody-littley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/config/DefaultConfiguration.java:56]:
```suggestion
    public static Configuration buildBasicConfiguration(@NonNull final List<Path> configurationPaths) throws IOException {
```

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/SocketConnection.java
Comment (cody-littley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/SocketConnection.java:72]: Similar to the reconnect classes from earlier in the PR, I suggest passing in a context (as the first argument) and then unpacking the configuration.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/reconnect/ReconnectLearner.java
Comment (cody-littley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/reconnect/ReconnectLearner.java:189]: If we used the paltform context, we could just store this class's version and we wouldn't have to worry about extracting data for the constructor.

**Comment by hendrikebbers** [comment]
@cody-littley agree 75% ;)

I would not use that pattern in common-module since we can have code here that should not depend on platform context in future but totally fine for platform-core.

But: I would not stop this PR. Once this PR is merged changing it to platform context is way easier than without this PR.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/config/DefaultConfiguration.java
Comment (timo0) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/config/DefaultConfiguration.java:39]: Done

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/jmh/java/com/swirlds/platform/benchmark/consensus/ConsensusBenchmark.java
Comment (alittley) [inline platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/jmh/java/com/swirlds/platform/benchmark/consensus/ConsensusBenchmark.java:75]: The previous TODO mentions using `scanAndRegisterAllConfigTypes`, but I don't understand the implications of that vs. `buildBasicConfiguration`

Could you clarify?

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/network/OutboundConnectionCreatorTest.java
Comment (alittley) [inline platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/network/OutboundConnectionCreatorTest.java:103]: there exists in this file the `getConfig` method. Any reason not to use it here?

Comment (timo0) [reply platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/network/OutboundConnectionCreatorTest.java:103]: Added, thanks!

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/SignedStateFileWriter.java
Comment (timo0) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/SignedStateFileWriter.java:185]: done, thanks!

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/reconnect/emergency/EmergencyReconnectTeacher.java
Comment (timo0) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/reconnect/emergency/EmergencyReconnectTeacher.java:68]: done, thanks!

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/SocketConnection.java
Comment (timo0) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/SocketConnection.java:109]: done, thanks!

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/SocketConnection.java
Comment (timo0) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/SocketConnection.java:72]: done, thanks!

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/connectivity/OutboundConnectionCreator.java
Comment (timo0) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/connectivity/OutboundConnectionCreator.java:71]: good catch, thanks!

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/connectivity/InboundConnectionHandler.java
Comment (timo0) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/connectivity/InboundConnectionHandler.java:74]: good catch, thanks!

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/config/DefaultConfiguration.java
Comment (timo0) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/config/DefaultConfiguration.java:39]: Done

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/merkle/synchronization/TeachingSynchronizer.java
Comment (timo0) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/merkle/synchronization/TeachingSynchronizer.java:124]: done, thanks!

**Review by alittley — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/config/DefaultConfiguration.java
Comment (imalygin) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/config/DefaultConfiguration.java:77]: I'd recommend using a log4j2 logger here.

Comment (timo0) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/config/DefaultConfiguration.java:77]: PCLI uses system.out across the board, not normal loggers - I think this should not block this PR since this makes up a separate discussions about whether sysout is the right choice or not.

**Review by hendrikebbers — COMMENTED** [review]
LGTM

**Pull request merged**
